### PR TITLE
Sensor for SSH connection in Tinom.

### PR DESCRIPTION
**Pull request by diarraa**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -32,15 +32,21 @@
 		<java.compiler.version>1.7</java.compiler.version>
 	</properties>
 
-	<dependencyManagement>
+<!-- 	<dependencyManagement> -->
```
Comment (vincent-zurczak) [inline pom.xml]: Remove completely.

File: src/main/java/org/occiware/tinom/Tinom.java [hunk 2]
```diff
@@ -22,4 +22,5 @@
  */
 public class Tinom {
 
+
```
Comment (vincent-zurczak) [inline src/main/java/org/occiware/tinom/Tinom.java]: Why this line break?

File: src/main/java/org/occiware/tinom/extensions/sample/nagiosconfig/SampleApp.java [hunk 3]
```diff
@@ -16,15 +16,13 @@
...
-import org.occiware.tinom.model.Sensor;
-
 /**
  * Sample application to generate a nagios hosts.cfg configuration file.
  * @author Pierre-Yves Gibello - Linagora
  */
 public class SampleApp {
 
-	public static void main(String[] args) {
+	/*public static void main(String[] args) throws JSchException, IOException {
```
Comment (vincent-zurczak) [inline src/main/java/org/occiware/tinom/extensions/sample/nagiosconfig/SampleApp.java]: Why is this method commented?
And main methods should not throw exceptions.

File: src/main/java/org/occiware/tinom/extensions/ssh/SSHconnectionSensor.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+import org.occiware.tinom.model.Sensor;
+
+import com.jcraft.jsch.Channel;
+import com.jcraft.jsch.ChannelExec;
+import com.jcraft.jsch.JSch;
+import com.jcraft.jsch.JSchException;
+import com.jcraft.jsch.Session;
+
+/**
+ * The class to use SSH connection with Tinom.
```
Comment (vincent-zurczak) [inline src/main/java/org/occiware/tinom/extensions/ssh/SSHconnectionSensor.java]: « A class to configure sensors with SSH. »

File: src/main/java/org/occiware/tinom/extensions/ssh/SSHconnectionSensor.java [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+			session.setPassword(password);
+			session.connect();
+
+			// Command execution
+			channel = session.openChannel("exec");
+			((ChannelExec)channel).setCommand(command);
+			channel.connect();
+
+		} catch (JSchException e) {
+
```
Comment (vincent-zurczak) [inline src/main/java/org/occiware/tinom/extensions/ssh/SSHconnectionSensor.java:103]: No sysout. Log instead.

File: src/main/java/org/occiware/tinom/extensions/ssh/SSHconnectionSensor.java [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+		} finally {
+
+			session.disconnect();
+			channel.disconnect();
+		}
+	}
+
+
+	/**
+	 * Sends a script file to agents in ssh and executes it.
```
Comment (vincent-zurczak) [inline src/main/java/org/occiware/tinom/extensions/ssh/SSHconnectionSensor.java]: Agents? Review the javadoc please.

File: src/main/java/org/occiware/tinom/extensions/ssh/SSHconnectionSensor.java [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+		String password = (String) prop.get(PASSWORD);
+		int port = Integer.parseInt( (String) prop.get(SSH_PORT) );
+
+		List<String> ips = Utils.readFileByLine(agentsIP);
+
+		// Prepare the command to execute
+		StringBuilder sb = new StringBuilder();
+		String scriptContent = Utils.readFileContent(script);
+
+		sb.append("bash <<ENDOFSCRIPT\n");
```
Comment (vincent-zurczak) [inline src/main/java/org/occiware/tinom/extensions/ssh/SSHconnectionSensor.java]: Is bash really necessary?
How about executing the script on Windows as an example? Can't we directly pass the script's content to the SSH library?

File: src/main/java/org/occiware/tinom/extensions/ssh/SSHconnectionSensor.java [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+		sb.append("ENDOFSCRIPT");
+		String command = sb.toString();
+
+		// Configure ssh
+		JSch jsch = new JSch();
+		jsch.setKnownHosts(knownHosts);
+		jsch.addIdentity(privKey);
+
+		for( String ip : ips) {
+			connectAndExecuteCommand(jsch, ip, username, password, command, port);
```
Comment (vincent-zurczak) [inline src/main/java/org/occiware/tinom/extensions/ssh/SSHconnectionSensor.java:155]: Add log traces before, during and after.

File: src/test/java/org/occiware/tinom/extensions/ssh/ToRunByHand.java [hunk 9]
```diff
@@ -0,0 +1,20 @@
+package org.occiware.tinom.extensions.ssh;
```
Comment (vincent-zurczak) [inline src/test/java/org/occiware/tinom/extensions/ssh/ToRunByHand.java]: Remove this file.
Unit tests might be better, with JUnit and EasyMock. Anyway, the code is simple enough...

File: src/test/java/org/occiware/tinom/extensions/ssh/ToRunByHand.java [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+
+import java.io.File;
+
+public class ToRunByHand {
```
Comment (vincent-zurczak) [inline src/test/java/org/occiware/tinom/extensions/ssh/ToRunByHand.java]: Add a new sample class showing how to use it.
And update the readme as well.

**Review by vincent-zurczak — CHANGES_REQUESTED** [review]

File: src/main/java/org/occiware/tinom/extensions/ssh/SSHconnectionSensor.java (hunk 6 above)
Comment (diarraa) [reply src/main/java/org/occiware/tinom/extensions/ssh/SSHconnectionSensor.java]: I wanted to say roboconf Agents.

**Pull request merged**